qemu/tests/qnum_change_reboot.py

```diff
@@ -0,0 +1,222 @@
+        nic_mq_info = session_serial.cmd_output_safe(mq_get_cmd)
+        queues_reg = re.compile(r"Combined:\s+(\d)", re.I)
+        queues_info = queues_reg.findall(" ".join(nic_mq_info.splitlines()))
+        if len(queues_info) != 2:
+            err_msg = "Oops, get guest queues info failed, "
+            err_msg += "make sure your guest support MQ.\n"
+            err_msg += "Check cmd is: '%s', " % mq_get_cmd
+            err_msg += "Command output is: '%s'." % nic_mq_info
+            test.cancel(err_msg)
+        return [int(x) for x in queues_info]
+
+    def enable_multi_queues(vm):
+        session_serial = vm.wait_for_serial_login(timeout=login_timeout)
+        error_context.context("Enable multi queues in guest.", logging.info)
+        for nic in vm.virtnet:
+            ifname = utils_net.get_linux_ifname(session_serial, nic.mac)
+            queues = int(nic.queues)
+            change_queues_number(ifname, queues)
+
+    def ping_test(dest_ip, ping_time, ping_lost_ratio):
+        _, output = utils_net.ping(dest=dest_ip, timeout=ping_time)
+        packets_lost = utils_test.get_loss_ratio(output)
+        if packets_lost > ping_lost_ratio:
+            err = " %s%% packages lost during ping. " % packets_lost
+            err += "Ping command log:\n %s" % "\n".join(
+                output.splitlines()[-3:])
+            test.fail(err)
+
+    def netperf_test():
+        """
+        Netperf stress test for nic option.
+        """
+        try:
+            n_server.start()
+            # Run netperf with message size defined in range.
+            netperf_test_duration = params.get_numeric("netperf_test_duration")
+            test_protocols = params.get("test_protocols", "TCP_STREAM")
+            netperf_output_unit = params.get("netperf_output_unit")
+            test_option = params.get("test_option", "")
+            test_option += " -l %s" % netperf_test_duration
+            if netperf_output_unit in "GMKgmk":
+                test_option += " -f %s" % netperf_output_unit
+            t_option = "%s -t %s" % (test_option, test_protocols)
+            n_client.bg_start(utils_net.get_host_ip_address(params),
+                              t_option,
+                              params.get_numeric("netperf_para_sessions"),
+                              params.get("netperf_cmd_prefix", ""),
+                              package_sizes=params.get("netperf_sizes"))
+            if utils_misc.wait_for(n_client.is_netperf_running, 10, 0, 1,
+                                   "Wait netperf test start"):
+                logging.info("Netperf test start successfully.")
+            else:
+                test.error("Can not start netperf client.")
+            utils_misc.wait_for(
+                lambda: not n_client.is_netperf_running(),
+                netperf_test_duration, 0, 5,
+                "Wait netperf test finish %ss" % netperf_test_duration)
+            time.sleep(5)
```

Comment:
Why need sleep 5 seconds?

Reply:
Thanks for your reminder,I will delete it. It is used to wait for the netperf process to start.Now this line 'wait_time = params.get_numeric("wait_bg_time")' can replace it.